history/history-network.md

```diff
@@ -236,45 +236,79 @@ content          = SSZ.serialize(block_header_with_proof)
+
+> Note: When the external oracle updates the tip of the chain causing previously valid headers to be invalidated, the client **SHOULD** not prune any non-finalized ephemeral headers until the node can rebuild a valid chain from latest finalized header to the HEAD of the chain.
 
 > Note: Clients **SHOULD** implement a mechanism to purge headers older than 8192 blocks from their content databases.
 
+##### Ephemeral Headers by `Find Content`
+
 ```python
 # Content and content key
 
-ephemeral_headers_key = Container(block_hash: Bytes32, ancestor_count: uint8)
+find_content_ephemeral_headers_key = Container(block_hash: Bytes32, ancestor_count: uint8)
 selector                 = 0x04
 
 BlockHeader              = ByteList[MAX_HEADER_LENGTH]
 ephemeral_header_payload = List(BlockHeader, limit=MAX_EPHEMERAL_HEADER_PAYLOAD)
 
 content                  = SSZ.serialize(ephemeral_header_payload)
-content_key              = selector + SSZ.serialize(ephemeral_headers_key)
+content_key              = selector + SSZ.serialize(find_content_ephemeral_headers_key)
 ```
 
-The `ephemeral_headers_key` encodes a request for headers anchored to the block
-hash indicated by `ephemeral_headers_key.block_hash`.  The
-`ephemeral_headers_key.ancestor_count` **MUST** be in the inclusive range
+The `find_content_ephemeral_headers_key` encodes a request for headers anchored to the block
+hash indicated by `find_content_ephemeral_headers_key.block_hash`. The
+`find_content_ephemeral_headers_key.ancestor_count` **MUST** be in the inclusive range
 0-255.
 
 The `ephemeral_header_payload` is an SSZ list of RLP encoded block header
-objects.  This object is subject to the following validity conditions.
+objects. This object is subject to the following validity conditions.
 
+* This content type **MUST** only be used for `Find Content` requests
 * The list **MAY** be empty which signals that the responding node was unable to fulfill the request.
-* The first element in the list **MUST** be the RLP encoded block header indicated by the `ephemeral_headers_key.block_hash` field from the content key.
+* The first element in the list **MUST** be the RLP encoded block header indicated by the `find_content_ephemeral_headers_key.block_hash` field from the content key.
 * Each element after the first element in the list **MUST** be the RLP encoded block header indicated by the `header.parent_hash` of the previous item from the list.
-* The list **SHOULD** contain no more than `ephemeral_headers_key.ancestor_count` items.
+* The list **SHOULD** contain no more than `find_content_ephemeral_headers_key.ancestor_count` items.
```

Comment:
The original spec also has this issue, but I think it should be `find_content_ephemeral_headers_key.ancestor_count + 1`, considering ancestor_count = 0 should return 1 header, the one with the defined blockhash.